app/static/js/volunteerProfile.js

```diff
@@ -86,17 +86,31 @@ $(document).ready(function(){
     });
   });
 
-
-  $(".backgroundCheck").change(function () { // Updates the Background check of a volunteer in the database
-    checkType = $(this).attr("id")
+  $(".savebtn").click(function () { // Updates the Background check of a volunteer in the database
+    let displayMsg = $("#displaySave")
+    let bgCheckType = $(this).data("id")
+    let bgDate = $("#" + bgCheckType + "_date").val()
+    let checkPassed = $("[data-id=" + bgCheckType + "]").val()
+
+    if (checkPassed == 0 && bgDate != '' ) {
+        displayMsg.html("Y/N<br>Empty!").addClass("text-danger")
+        setTimeout(function() {displayMsg.html("").removeClass("text-danger")}, 2000)
+        return
+    }else if(bgDate == '' && checkPassed != 0){
+        displayMsg.html("Date<br>Empty!").addClass("text-danger")
+        setTimeout(function() {displayMsg.html("").removeClass("text-danger")}, 2000)
+        return
+    }
+    else{
+    displayMsg.html("Saved!").addClass("text-success")
+    setTimeout(function() {displayMsg.html("").removeClass("text-success")}, 2000)
+    }
```

Comment:
Let's consolidate this "display a message and then clear it after 2 seconds" code into a function. That will make the code a little clearer, and if we need to modify it in the future it will be easier.

Reply:
resolved